convert2rhel/actions/system_checks/check_firewalld_availability.py

```diff
@@ -0,0 +1,133 @@
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+import os
+import re
+
+from convert2rhel import actions, systeminfo
+from convert2rhel.systeminfo import system_info
+
+
+logger = logging.getLogger(__name__)
+
+# Path to the firewalld config file.
+FIREWALLD_CONFIG_FILE = "/etc/firewalld/firewalld.conf"
+
+# Regex that identify if CleanupModulesOnExit is set to either yes or true
+# considering blank spaces in between the value and case insensitive.
+CLEANUP_MODULES_ON_EXIT_REGEX = re.compile(r"(?i)CleanupModulesOnExit\s*=\s*(yes|true)")
+
+
+def _is_modules_cleanup_enabled():
+    """Verify firewalld modules cleanup config is enabled.
+
+    :returns: Whether or not the CleanupModulesOnExit is set to true in
+        firewalld config.
+    :rtype: bool
+    """
+    # Return false if the config file does not exist. Either it means that
+    # firewalld is not installed or the config file was removed.
+    if not os.path.exists(FIREWALLD_CONFIG_FILE):
+        logger.debug("%s does not exist." % FIREWALLD_CONFIG_FILE)
+        return False
+
+    contents = []
+    with open(FIREWALLD_CONFIG_FILE, mode="r") as handler:
+        contents = [line.strip() for line in handler.readlines() if line.strip()]
+
+    # Contents list is empty for some reason, better to assume that there
+    # is no content in the file that was read.
+    if not contents:
+        logger.debug("%s is empty." % FIREWALLD_CONFIG_FILE)
+        return True
+
+    # If the CleanupModulesOnExit is not present inside the contents list, we
+    # can return True since the default behavior for firewalld is to consider
+    # CleanupModulesOnExit as true. Ignoring # and ; as they are ignored in the
+    # config parser for firewalld as well:
+    # https://github.com/firewalld/firewalld/blob/46d54dcbdff94423686d67befc78ca8d601fce60/src/firewall/core/io/firewalld_conf.py#L85
+    option_present = any(
+        item.strip().startswith("CleanupModulesOnExit") for item in contents if not item.startswith(("#", ";"))
```

Comment:
```suggestion
        item.strip().startswith("CleanupModulesOnExit") for item in contents
```

I think that that the `if not item.startswith(("#", ";"))` is not necessary because if `item.strip().startswith("CleanupModulesOnExit")` returns True, it means that there's nothing before the CleanupModulesOnExit.

Reply:
But the `if not item.startswith(("#", ";"))` is to prevent that we are returning True when we find a line with a comment, in this case, we will only proceed on checking that if the line does not have either `#` or `;`, meaning that lines with this are skippable.